src/armor_layers.cpp

```diff
@@ -512,222 +618,168 @@ void player::sort_armor()
                 return;
             }
         }
-        werase( w_sort_cat );
-        werase( w_sort_left );
-        werase( w_sort_middle );
-        werase( w_sort_right );
-        werase( w_encumb );
-
-        // top bar
-        wprintz( w_sort_cat, c_white, _( "Sort Armor" ) );
-        wprintz( w_sort_cat, c_yellow, "  << %s >>", armor_cat[tabindex] );
-        right_print( w_sort_cat, 0, 0, c_white, string_format(
-                         _( "Press %s for help. Press %s to change keybindings." ),
-                         ctxt.get_desc( "USAGE_HELP" ),
-                         ctxt.get_desc( "HELP_KEYBINDINGS" ) ) );
-
-        // Create ptr list of items to display
-        tmp_worn.clear();
-        if( tabindex == num_bp ) { // All
-            for( auto it = worn.begin(); it != worn.end(); ++it ) {
-                tmp_worn.push_back( it );
-            }
-        } else { // bp_*
-            body_part bp = static_cast<body_part>( tabindex );
-            for( auto it = worn.begin(); it != worn.end(); ++it ) {
-                if( it->covers( bp ) ) {
-                    tmp_worn.push_back( it );
+
+        // set up data
+        for( size_t bp = 0; bp < num_bp; bp++ ) {
+            items_by_category[bp].clear();
+        }
+        for( auto elem_it = worn.begin(); elem_it != worn.end(); ++elem_it ) {
+            for( size_t bp = 0; bp < num_bp; bp++ ) {
```

Comment:
```suggestion
            for( const body_part bp : all_body_parts ) {
```
So you don't have to cast the `size_t` in to a `body_part`.